### PR TITLE
Task/http binding

### PR DESCRIPTION
**Pull request by dmoranj**
Implements 110

### CODE REVIEW

File: lib/bindings/HTTPBinding.js [hunk 1]
```diff
@@ -0,0 +1,283 @@
...
+    };
+
+    request(options, function(error, response, body) {
+        if (error || response.statusCode !== 200) {
+            callback(new errors.HTTPCommandResponseError(response.statusCode, error));
+        } else {
+            process.nextTick(commandHandler.updateCommand.bind(null, apiKey, device, body, callback));
+        }
+
+        callback();
```
Comment (gtorodelvalle) [inline lib/bindings/HTTPBinding.js]: This `callback()` invocation may lead to unexpected behaviours as a consequence of the `callback()` being invoked twice and not with the expected arguments, doesn't it?

File: test/deviceProvisioning/provisionDeviceHTTP.json [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+          "type": "celsius"
+        },
+        {
+          "name": "humidity",
+          "type": "degrees"
+        }
+      ]
+    }
+  ]
+}
```
Comment (gtorodelvalle) [inline test/deviceProvisioning/provisionDeviceHTTP.json]: Final blank line appreciated... ;)

File: lib/bindings/HTTPBinding.js (hunk 1 above)
Comment (dmoranj) [inline lib/bindings/HTTPBinding.js]: Agree. Fixed in 26d89d1

**Comment by gtorodelvalle** [comment]
I noticed that some of the tests files have `-test` as suffix and other ones `_test`, just in case you want to fix that... :) Everything should work regarding the automatic unit tests, though... :)

File: lib/bindings/HTTPBinding.js [hunk 3]
```diff
@@ -117,9 +117,8 @@ function executeCommand(apiKey, device, serializedPayload, callback) {
             callback(new errors.HTTPCommandResponseError(response.statusCode, error));
         } else {
             process.nextTick(commandHandler.updateCommand.bind(null, apiKey, device, body, callback));
+            callback();
```
Comment (gtorodelvalle) [inline lib/bindings/HTTPBinding.js]: I think you have to remove it from here too since already included into the previous function call as the callback.

Comment (dmoranj) [inline lib/bindings/HTTPBinding.js]: Fixed in d8d69bd. Again

**Comment by gtorodelvalle** [comment]
LGTM ;)

File: lib/bindings/HTTPBinding.js (hunk 3 above)
Comment (dmoranj) [inline lib/bindings/HTTPBinding.js]: And yet another time 64f5007. Loooong friday.

**Pull request merged**